source/ui/x11/X11Window.ooc

```diff
@@ -52,8 +64,34 @@ X11Window: class extends NativeWindow {
 	}
 	resize: func (size: IntSize2D) { XResizeWindow(this display, this backend, size width, size height) }
 	free: override func {
+		if (this _xImage)
+			XDestroyImage(this _xImage)
+		XFreeGC(this display, this _x11GraphicsContext)
+		XDestroyWindow(this display, this backend)
 		XCloseDisplay(this display)
 		super()
 	}
+	draw: func (image: RasterYuv420Semiplanar) {
+		if (this _cacheSize != image size) {
+			if (this _xImage)
+				XDestroyImage(this _xImage)
+			imageData := gc_malloc(image width * image height * 4)
+			this _xImage = XCreateImage(this display, CopyFromParent, this _displayDepth, ZPixmap, 0, imageData, image width, image height, 8, 0)
+			this _cacheSize = image size
+		}
+		for (row in 0 .. image height ) {
+			for (column in 0 .. image width) {
+				pixel := 0 as ULong
+				pixelData := pixel& as UChar*
+				color := image[column, row] toBgr()
+				pixelData@ = color blue as UChar
+				(pixelData + 1)@ = color green as UChar
+				(pixelData + 2)@ = color red as UChar
+				XPutPixel(this _xImage, column, row, pixel)
+			}
+		}
+		XClearWindow(this display, this backend)
```

Comment:
I will fix this as soon as https://github.com/cogneco/ooc-kean/pull/665 is merged. This is on top of my TODO list in my head because it looks horribly :)